test/mpi/test_mpi.cc

```diff
@@ -111,12 +139,23 @@ UccTestMpi::~UccTestMpi()
     for (auto &t : teams) {
         destroy_team(t);
     }
+    for (auto &t : onesided_teams) {
+        destroy_team(t);
+    }
+    if (onesided_buffers) {
+        for (auto i = 0; i < UCC_TEST_N_MEM_SEGMENTS; i++) {
+            ucc_free(onesided_buffers[i]);
+        }
+    }
     UCC_CHECK(ucc_context_destroy(ctx));
+    UCC_CHECK(ucc_context_destroy(onesided_ctx));
     UCC_CHECK(ucc_finalize(lib));
+    UCC_CHECK(ucc_finalize(onesided_lib));
```

Comment:
btw, we should be able to create 2 contexts from 1 lib object, right?

Reply:
yes, we should be able to, but for some reason reason only ucc_lib_config_read() pulls in the environment variable change for UCC_TL_UCP_TUNE. that's why i'm creating two. please let me know if I'm doing something wrong there..